src/Widgets/Modal/SubmitProposal.hs

```diff
@@ -91,11 +92,26 @@ saveScenarioContent renderingApi scenarioB submitPopupUrlE = do
                $ makePut <$> submitUrlB <*> imgurlB <*> scContentB <@> submitWithDescrE
 
     performEvent_ . ffor responseE $ \eresponse -> case eresponse of
-      Left (JSError err) -> showUserMessage . SingleMsg $
-              "An error happened when submitting the design.\n"
-           <> "Please, save your design to your computer and contact administrators.\n"
-           <> "Error: " <> err
-      Right _  -> showUserMessage $ SingleMsg $
+      Left (JSError err) -> do
+        sc <- sample scenarioB
+        prepareDownloadScenarioContent sc
+        showUserPopup . SingleMsg $
+              "<div>"
```

Comment:
Let's put the whole thing into hamlet splice (without qhtml, because we need to put dynamic arguments in there).

Reply:
ok, figured out how to do it...